### PR TITLE
chore: display TinEye search results

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves []

### Description
Added a screen that displays TinEye search results. Don't pay attention to the design, as this screen is more intended for debug/development purposes and most likely this screen will be removed in the future

### Demo
https://user-images.githubusercontent.com/3513494/168392211-3f2878dc-10dc-4d5e-ad3e-a988b4979b4c.mp4

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- display TinEye search results - dimatretyak

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: /docs/adding_state_migrations.md
[feature flag]: /docs/developing_a_feature.md
[docs]: /docs/README.md

### CODE REVIEW

File: src/app/Scenes/ReverseSearchImageResults/ReverseSearchImageResults.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    } catch (error) {
+      console.error(error)
+      Alert.alert("Something went wrong", (error as Error).message)
+    } finally {
+      setFetching(false)
+    }
+  }
+
+  const isEmptyResults = Array.isArray(results) && results.length === 0
+  const hasResults = Array.isArray(results) && results.length > 0
```
Comment (gkartalis) [inline src/app/Scenes/ReverseSearchImageResults/ReverseSearchImageResults.tsx:90]: **suggestion:** I think we can use `isEmpty` from [lodash](https://lodash.com/docs/4.17.15#isEmpty) for this. Let me know your thoughts
```suggestion
  const hasResults = !lodash.isEmpty(results)
```

File: src/app/Scenes/ReverseSearchImageResults/ReverseSearchImageResults.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            </Text>
+          )}
+
+          {!!hasResults &&
+            results.map((result) => (
+              <ReverseSearchImageResultItemFragmentContainer
+                key={result.artwork?.internalID}
+                item={result}
+              />
+            ))}
```
Comment (gkartalis) [inline src/app/Scenes/ReverseSearchImageResults/ReverseSearchImageResults.tsx:113]: and then we can use the `hasResults` to conditionally render the noResults/list of results
```suggestion
          {!!hasResults ?
          (<>
          results.map((result) => (
              <ReverseSearchImageResultItemFragmentContainer
                key={result.artwork?.internalID}
                item={result}
              />
            ))
            </>)
            :
            <Text textAlign="center" my={2}>
              No matches
            </Text>
       }
            
```

**Review by gkartalis — COMMENTED** [review]
Looking good! Left one suggestion but other than that thanks! 🎉

File: src/app/Scenes/ReverseSearchImageResults/ReverseSearchImageResults.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/app/Scenes/ReverseSearchImageResults/ReverseSearchImageResults.tsx:113]: In this case, we will have "No matches" as soon as the user opened the screen. I wanted to render the search results or "No matches" only after the user selected the image

**Review by gkartalis — APPROVED** [review]

**Pull request merged**